File: 2026-2/sesion-de-clase-08/softprog/softprog-modelo/src/main/java/pe/edu/pucp/softprog/modelo/rrhh/Area.java
package pe.edu.pucp.softprog.modelo.rrhh;

import pe.edu.pucp.softprog.modelo.Registro;

public class Area extends Registro {
    private String nombre;

    public Area() {
    }

    public Area(final Area area) {
        if (area == null) {
            throw new IllegalArgumentException("area no puede ser nulo");
        }

        super(area);
        setNombre(area.getNombre());
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        if (nombre == null || nombre.isEmpty()) {
            throw new IllegalArgumentException("nombre no puede ser nulo o vacío");
        }
        this.nombre = nombre;
    }

    @Override
    public String toString() {
        return super.toString() + String.format("\t%-20s", nombre);
    }
}

File: 2026-2/sesion-de-clase-11/SoftProg/SoftProg.Modelo/RRHH/Area.cs
namespace SoftProg.Modelo.RRHH {
    public class Area : Registro {
        public Area() { }

        public Area(Area area) : base(area) {
            Nombre = area.Nombre;
        }

        public string Nombre { 
            get; 
            set {
                ArgumentNullException.ThrowIfNullOrEmpty(value);
                field = value;
            }
        }

        public override string ToString() {
            return base.ToString() + $"Nombre: {Nombre}";
        }
    }
}
